File: CinemaReservations/CSharp/CinemaAllocations.Domain/SeatsAllocated.cs
using System.Collections.Generic;

namespace CinemaAllocations.Domain
{
    public class SeatsAllocated
    {
        public int PartyRequested { get; }
        
        public List<Seat> ReservedSeats { get; }

        public SeatsAllocated(List<Seat> reservedSeats, int partyRequested)
        {
            ReservedSeats = reservedSeats;
            PartyRequested = partyRequested;
        }
    }
}

File: CinemaReservations/Java/CinemaAllocationsDomain/src/main/java/com/sdd/cinemaallocations/SeatsAllocated.java
package com.sdd.cinemaallocations;

import lombok.Value;
import lombok.experimental.NonFinal;

import java.util.Comparator;
import java.util.List;
import java.util.stream.Collectors;

@Value
@NonFinal
public class SeatsAllocated {
    int partyRequested;
    List<Seat> reservedSeats;

    public List<String> seatNames() {
        return reservedSeats.stream().sorted(Comparator.comparing(Seat::number)).map(Seat::toString).collect(Collectors.toList());
    }
}
